deduce/utility.py

```diff
@@ -321,3 +336,34 @@ def read_list(list_name, encoding='utf-8', lower=False,
         data_nodoubles = list(set(data))
 
     return data_nodoubles
+
+def parse_tag(tag: str) -> tuple:
+    """
+    Parse a Deduce-style tag into its tag proper and its text. Does not handle nested tags
+    :param tag: the Deduce-style tag, for example, <VOORNAAMONBEKEND Peter>
+    :return: the tag type and text, for example, ("VOORNAAMONBEKEND", "Peter")
+    """
+    split_ix = tag.index(" ")
+    return tag[1:split_ix], tag[split_ix+1:len(tag)-1]
+
+def get_annotations(annotated_text: str, tags: list, n_leading_whitespaces=0) -> list:
+    """
+    Find structured annotations from tags, with indices pointing to the original text. Does not handle nested tags
+    :param annotated_text: the annotated text
+    :param tags: the tags found in the text
+    :param n_leading_whitespaces: the number of leading whitespaces in the raw text
+    :return: the annotations with indices corresponding to the original (raw) text
+    """
+    ix = 0
+    annotations = []
+    raw_text_ix = n_leading_whitespaces
+    for tag in tags:
+        tag_ix = annotated_text.index(tag, ix) - ix
```

Comment:
This is because deduce.annotate_text removes leading whitespaces. So, in order to reconstruct the position of the tagged text in the original text, I need to know the number of leading whitespaces in the original text